From b1cd6fb95078a37dc6d7905568b38e47350e24c2 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sun, 13 Jul 2014 19:31:03 +1000
Subject: [PATCH 2/4] metadata: Properly separate TV title/subtitle search
 from inetref/subtitle search
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This prevent likely errors of parsing a title as an inetref, in particular if it contains a ‘:’

(cherry picked from commit a013a2e0b212526994263e3489a90bf5a85f5265)
---
 mythtv/libs/libmythmetadata/metadatadownload.cpp |    1 +
 mythtv/libs/libmythmetadata/metadatagrabber.cpp  |   15 +++++++++++++++
 mythtv/libs/libmythmetadata/metadatagrabber.h    |    1 +
 3 files changed, 17 insertions(+)

diff --git a/mythtv/libs/libmythmetadata/metadatadownload.cpp b/mythtv/libs/libmythmetadata/metadatadownload.cpp
index d085549..a84d6d2 100644
--- a/mythtv/libs/libmythmetadata/metadatadownload.cpp
+++ b/mythtv/libs/libmythmetadata/metadatadownload.cpp
@@ -587,6 +587,7 @@ MetadataLookupList MetadataDownload::handleVideoUndetermined(MetadataLookup *loo
     else
     {
         list = grabber.SearchSubtitle(lookup->GetInetref(),
+                                      lookup->GetTitle(),
                                       lookup->GetSubtitle(), lookup, false);
     }
 
diff --git a/mythtv/libs/libmythmetadata/metadatagrabber.cpp b/mythtv/libs/libmythmetadata/metadatagrabber.cpp
index bb30593..5656e76 100644
--- a/mythtv/libs/libmythmetadata/metadatagrabber.cpp
+++ b/mythtv/libs/libmythmetadata/metadatagrabber.cpp
@@ -514,6 +514,21 @@ MetadataLookupList MetaGrabberScript::SearchSubtitle(const QString &title,
     return RunGrabber(args, lookup, passseas);
 }
 
+MetadataLookupList MetaGrabberScript::SearchSubtitle(const QString &inetref,
+                        const QString &title, const QString &subtitle,
+                        MetadataLookup *lookup, bool passseas)
+{
+    (void)title;
+    QStringList args;
+    SetDefaultArgs(args);
+
+    args << "-N"
+         << CleanedInetref(inetref)
+         << subtitle;
+
+    return RunGrabber(args, lookup, passseas);
+}
+
 MetadataLookupList MetaGrabberScript::LookupData(const QString &inetref,
                         MetadataLookup *lookup, bool passseas)
 {
diff --git a/mythtv/libs/libmythmetadata/metadatagrabber.h b/mythtv/libs/libmythmetadata/metadatagrabber.h
index 7d64e17..0a54d39 100644
--- a/mythtv/libs/libmythmetadata/metadatagrabber.h
+++ b/mythtv/libs/libmythmetadata/metadatagrabber.h
@@ -72,6 +72,7 @@ class META_PUBLIC MetaGrabberScript : public QObject
     bool                Test(void);
     MetadataLookupList  Search(const QString &title, MetadataLookup *lookup, bool passseas=true);
     MetadataLookupList  SearchSubtitle(const QString &title, const QString &subtitle, MetadataLookup *lookup, bool passseas=true);
+    MetadataLookupList  SearchSubtitle(const QString &inetref, const QString &title, const QString &subtitle, MetadataLookup *lookup, bool passseas=true);
     MetadataLookupList  LookupData(const QString &inetref, MetadataLookup *lookup, bool passseas=true);
     MetadataLookupList  LookupData(const QString &inetref, int season, int episode, MetadataLookup *lookup, bool passseas=true);
     MetadataLookupList  LookupCollection(const QString &collectionref, MetadataLookup *lookup, bool passseas=true);
-- 
1.7.10.2

